### PR TITLE
test that linkage errors don't increase

### LINKED ISSUE #654: Add test for BOM that establishes maximum linkage errors to prevent regressions

### PR DESCRIPTION
**Pull request by elharo**
fix #654

### CODE REVIEW

File: boms/integration-tests/pom.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.google.cloud.tools</groupId>
+      <artifactId>dependencies</artifactId>
+      <version>0.2.1-SNAPSHOT</version>
```
Comment (suztomo) [inline boms/integration-tests/pom.xml]: Does this version get updated when `prepare_release.sh` runs in the root directory?
- If yes, no problem.
- If no, how about adding the parent pom as below?

```
  <parent>
    <groupId>com.google.cloud.tools</groupId>
    <artifactId>dependencies-parent</artifactId>
    <version>0.2.1-SNAPSHOT</version>
  </parent>
...
<dependencies>
    <dependency>
      <groupId>${project.groupId}</groupId>
      <artifactId>dependencies</artifactId>
      <version>${project.version}</version>
    </dependency>
   ...
```

**Comment by elharo** [comment]
Problems on CircleCI don't appear to be flake related:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test (default-test) on project integration-tests: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test failed: The forked VM terminated without saying properly goodbye. VM crash or System.exit called ? -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test (default-test) on project integration-tests: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test failed: The forked VM terminated without saying properly goodbye. VM crash or System.exit called ?

**Comment by suztomo** [comment]
> [INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ integration-tests ---
> ...
> Error: Could not find or load main class org.apache.maven.surefire.booter.ForkedBooter
https://circleci.com/gh/GoogleCloudPlatform/cloud-opensource-java/1701?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Same as #197 ?

**Comment by elharo** [comment]
Probably not exaclty. That symptom has a lot of causes.

File: boms/integration-tests/pom.xml (hunk 1 above)
Comment (elharo) [reply boms/integration-tests/pom.xml]: Yes, it does. I just tested it:

```
 <groupId>com.google.cloud</groupId>
  <artifactId>integration-tests</artifactId>
  <version>1.2.0-SNAPSHOT</version>

  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <maven.compiler.source>1.8</maven.compiler.source>
    <maven.compiler.target>1.8</maven.compiler.target>
  </properties>

  <dependencies>
    <dependency>
      <groupId>com.google.cloud.tools</groupId>
      <artifactId>dependencies</artifactId>
      <version>5.6.8-SNAPSHOT</version>
    </dependency>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>4.12</version>
      <scope>test</scope>
    </dependency>
```

Comment (elharo) [reply boms/integration-tests/pom.xml]: Yes, it does.
